### PR TITLE
Vinylslider dB Scale 

### PR DESCRIPTION
**Pull request by daschuer**
see commits.

### CODE REVIEW

File: src/dlgprefvinyl.cpp [hunk 1]
```diff
@@ -185,6 +187,8 @@ void DlgPrefVinyl::slotResetToDefaults() {
     LeadinTime4->setText(QString("%1").arg(MIXXX_VINYL_SERATOCV02VINYLSIDEA_LEADIN));
     SignalQualityEnable->setChecked(true);
     VinylGain->setValue(0);
+    slotUpdateVinylGain();
+    qDebug() << "reset";
```
Comment (ywwg) [inline src/dlgprefvinyl.cpp]: stray debug

File: src/dlgprefvinyl.cpp [hunk 2]
```diff
@@ -252,7 +256,12 @@ void DlgPrefVinyl::slotUpdate() {
             (bool)config->getValueString(ConfigKey(VINYL_PREF_KEY, "show_signal_quality")).toInt());
 
     //set vinyl control gain
-    VinylGain->setValue(config->getValueString(ConfigKey(VINYL_PREF_KEY, "gain")).toInt());
+    double ratioGain = config->getValueString(ConfigKey(VINYL_PREF_KEY, "gain")).toDouble();
+    double dbGain = ratio2db(ratioGain);
+    VinylGain->setValue((int)(dbGain + 0.5));
+    slotUpdateVinylGain();
+
+    qDebug() << "reset2" << ratioGain << dbGain;
```
Comment (ywwg) [inline src/dlgprefvinyl.cpp]: etc

File: src/dlgprefvinyldlg.ui [hunk 3]
```diff
@@ -593,65 +593,65 @@
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
-       <layout class="QHBoxLayout" name="horizontalLayout_2">
+       <widget class="QLabel" name="textLabel1_3">
+        <property name="text">
+         <string>Turntable Preamp</string>
```
Comment (ywwg) [inline src/dlgprefvinyldlg.ui]: Use whole words, and in other parts of the code we say "boost", so maybe "Turntable input signal boost"

File: src/dlgprefvinyl.cpp [hunk 4]
```diff
@@ -366,13 +375,21 @@ void DlgPrefVinyl::VinylTypeSlotApply()
...
+    config->set(ConfigKey(VINYL_PREF_KEY, "gain"), ConfigValue(QString::number(ratioGain)));
 
     // Update the ControlObject...
-    ControlObject::set(ConfigKey(VINYL_PREF_KEY, "gain"), VinylGain->value());
+    ControlObject::set(ConfigKey(VINYL_PREF_KEY, "gain"), ratioGain);
+}
+
+void DlgPrefVinyl::slotUpdateVinylGain() {
+    int value = VinylGain->value();
+    textLablePreampCurrent->setText(
```
Comment (ywwg) [inline src/dlgprefvinyl.cpp]: misspelled.

File: src/dlgprefvinyl.cpp [hunk 5]
```diff
@@ -366,13 +375,21 @@ void DlgPrefVinyl::VinylTypeSlotApply()
...
 }
 
-void DlgPrefVinyl::VinylGainSlotApply() {
-    qDebug() << "in VinylGainSlotApply()" << "with gain:" << VinylGain->value();
+void DlgPrefVinyl::slotVinylGainApply() {
+    int dBGain = VinylGain->value();
+    qDebug() << "in VinylGainSlotApply()" << "with gain:" << dBGain << "dB";
     // Update the config key...
-    config->set(ConfigKey(VINYL_PREF_KEY, "gain"), ConfigValue(VinylGain->value()));
+    double ratioGain = db2ratio((double)dBGain);
```
Comment (ywwg) [inline src/dlgprefvinyl.cpp]: c++-style cast

**Comment by daschuer** [comment]
fixed!

File: src/dlgprefnovinyldlg.ui [hunk 6]
```diff
@@ -6,398 +6,15 @@
...
-          <height>40</height>
-         </size>
-        </property>
-       </spacer>
-      </item>
-     </layout>
-    </widget>
-   </item>
-   <item row="6" column="0" colspan="2">
+   <item row="0" column="0" colspan="2">
```
Comment (ywwg) [inline src/dlgprefnovinyldlg.ui]: I confirmed that this change is ok (built and ran vinylcontrol=0)

File: src/dlgprefvinyl.cpp [hunk 7]
```diff
@@ -89,8 +89,10 @@ DlgPrefVinyl::DlgPrefVinyl(QWidget * parent, VinylControlManager *pVCMan,
     ComboBoxVinylSpeed4->addItem(MIXXX_VINYL_SPEED_33);
     ComboBoxVinylSpeed4->addItem(MIXXX_VINYL_SPEED_45);
 
-    // TODO: Convert gain to logarithmic and db visual scale.
-    connect(VinylGain, SIGNAL(sliderReleased()), this, SLOT(VinylGainSlotApply()));
+    connect(VinylGain, SIGNAL(sliderReleased()),
+            this, SLOT(slotVinylGainApply()));
+    connect(VinylGain, SIGNAL(valueChanged(int)),
+            this, SLOT(slotUpdateVinylGain()));
     //connect(ComboBoxDeviceDeck1, SIGNAL(currentIndexChanged()), this, SLOT(()));
```
Comment (ywwg) [inline src/dlgprefvinyl.cpp]: might as well blow away this dead code

File: src/dlgprefvinyl.cpp [hunk 8]
```diff
@@ -252,7 +255,10 @@ void DlgPrefVinyl::slotUpdate() {
             (bool)config->getValueString(ConfigKey(VINYL_PREF_KEY, "show_signal_quality")).toInt());
 
     //set vinyl control gain
-    VinylGain->setValue(config->getValueString(ConfigKey(VINYL_PREF_KEY, "gain")).toInt());
+    double ratioGain = config->getValueString(ConfigKey(VINYL_PREF_KEY, "gain")).toDouble();
```
Comment (ywwg) [inline src/dlgprefvinyl.cpp]: could constify these values

File: src/dlgprefvinyldlg.ui [hunk 9]
```diff
@@ -593,65 +593,65 @@
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
-       <layout class="QHBoxLayout" name="horizontalLayout_2">
+       <widget class="QLabel" name="textLabel1_3">
+        <property name="text">
+         <string>Turntable input signal boost</string>
```
Comment (ywwg) [inline src/dlgprefvinyldlg.ui]: sorry, actually I should have asked for the to be Title Capitalization: Turntable Input Signal Boost

**Comment by ywwg** [comment]
couple more comments, some optional, then LGTM

**Comment by daschuer** [comment]
Thank you for review!

**Pull request merged**